sdk/servicebus/service-bus/src/serviceBusAtomManagementClient.ts

```diff
@@ -0,0 +1,547 @@
+   * @param {object}             requestBody                                         The resource handler.
+   * @param {ResourceSerializer} serializer                                          The XML serializer to use.
+   * @return {Promise<HttpOperationResponse>}
+   */
+  private async _listResources(
+    path: string,
+    requestBody: any,
+    serializer: ResourceSerializer
+  ): Promise<HttpOperationResponse> {
+    let webResource: WebResource = new WebResource();
+
+    webResource.method = "GET";
+
+    const queryParams: any = {};
+    if (requestBody) {
+      if (requestBody.skip) {
+        queryParams["$skip"] = requestBody.skip;
+      }
+      if (requestBody.top) {
+        queryParams["$top"] = requestBody.top;
+      }
+    }
+
+    webResource.url = this.getUrl(path, queryParams);
+
+    console.log(webResource.url);
+
+    const atomXmlOperationSpec: AtomXmlOperationSpec = {
+      serializer: serializer,
+      shouldParseResponse: true
+    };
+    webResource.atomXmlOperationSpec = atomXmlOperationSpec;
+
+    return this.sendRequest(webResource);
+  }
+
+  /**
+   * Lists resources.
+   *
+   * @param {string}             path                                                The resource path.
+   * @return {Promise<HttpOperationResponse>}
+   */
+  private async _deleteResource(path: any): Promise<HttpOperationResponse> {
+    let webResource: WebResource = new WebResource();
+
+    webResource.method = "DELETE";
+    webResource.url = this.getUrl(path);
+
+    return this.sendRequest(webResource);
+  }
+
+  private getUrl(path: string, queryParams?: any) {
+    const baseUri = `https://${this.endpoint}.servicebus.windows.net/${path}`;
+
+    const requestUrl: URLBuilder = URLBuilder.parse(baseUri);
+    requestUrl.setQueryParameter(`api-version`, `2017-04`);
+
+    if (queryParams) {
+      for (let key of Object.keys(queryParams)) {
+        requestUrl.setQueryParameter(key, queryParams[key]);
+      }
+    }
+
+    return requestUrl.toString();
+  }
+  private static parseConnectionString(connectionString: string): any {
+    const output: { [k: string]: string } = {};
+    const parts = connectionString.trim().split(";");
+
+    for (let part of parts) {
+      part = part.trim();
+
+      if (part === "") {
+        // parts can be empty
+        continue;
+      }
+
+      const splitIndex = part.indexOf("=");
+      if (splitIndex === -1) {
+        throw new Error(
+          "Connection string malformed: each part of the connection string must have an `=` assignment."
+        );
+      }
+
+      const key = part.substring(0, splitIndex).trim();
+      if (key === "") {
+        throw new Error("Connection string malformed: missing key for assignment");
+      }
+
+      const value = part.substring(splitIndex + 1).trim();
+
+      output[key] = value;
+    }
+
+    return output as any;
+  }
+
+  private static getSubscriptionPath(topic: string, subscription: string) {
```

Comment:
Any reason to have `getSubscriptionPath` and `getRulePath` as static methods?
These are private methods. Usually we would want to have **private** static methods so that they can be used by other static methods in the same class.

But this is not the case here. These should be private instance methods